File: Application/DTOs/MedicationCreateDto.cs
using System.ComponentModel.DataAnnotations;
using Domain;

namespace Application.DTOs;

public record class MedicationCreateDto
(
    [Required]
    Guid FirstAidKitId,

    [Required]
    [MaxLength(200)]
    string Name,

    [Range(1, int.MaxValue)]
    int Quantity,

    [Range(1, int.MaxValue)]
    int MinimumQuantity,

    MeasurementUnit Unit,

    [Required]
    DateTime ExpirationDate
);

File: frontend/lib/data/dtos/medication_create_dto.dart

import 'package:json_annotation/json_annotation.dart';
import 'package:frontend/data/dtos/measurement_unit.dart';

part 'medication_create_dto.g.dart';

@JsonSerializable()
class MedicationCreateDto {
  final String firstAidKitId;
  final String name;
  final int quantity;
  final int minimumQuantity;
  final MeasurementUnit unit;
  final DateTime expirationDate;

  MedicationCreateDto({
    required this.firstAidKitId,
    required this.name,
    required this.quantity,
    required this.minimumQuantity,
    required this.unit,
    required this.expirationDate,
  });

  factory MedicationCreateDto.fromJson(Map<String, dynamic> json) => _$MedicationCreateDtoFromJson(json);
  Map<String, dynamic> toJson() => _$MedicationCreateDtoToJson(this);
}